### PR TITLE
Fix Redis Query for RediSearch 2.8+

### PR DESCRIPTION
**Pull request by JoshVanL**
In RediSearch 2.8 the returned data structure of an `FT.SEARCH` changed, breaking the state Query API.

PR updates the redis Query function to parse both the old and new returned structures. Adds a query certification test.

https://docs.redis.com/latest/stack/release-notes/redisearch/redisearch-2.8-release-notes/

### CODE REVIEW

File: tests/certification/state/redis/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 version: "3.3"
 services:
   redis:
-    image: 'redislabs/redisearch:latest'
+    image: 'redislabs/redisearch:edge'
```
Comment (JoshVanL) [inline tests/certification/state/redis/docker-compose.yml:4]: edge includes the later 2.8 redis search module.

File: tests/certification/state/redis/docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
     ports:
       - '6379:6379'
-    command: redis-server
```
Comment (JoshVanL) [inline tests/certification/state/redis/docker-compose.yml:7]: Adding this command prevented both the search and json modules from being loaded, which are needed for the query API.

File: tests/certification/state/redis/docker-compose.yml (hunk 1 above)
Comment (ItalyPaleAle) [reply tests/certification/state/redis/docker-compose.yml:4]: Please don't pin to non-stable builds. I trust that the bug was there :)

File: state/redis/redis_query.go [hunk 3]
```diff
@@ -295,10 +295,49 @@ func (q *Query) execute(ctx context.Context, client rediscomponent.RedisClient)
 	if err != nil {
 		return nil, "", err
 	}
-	arr, ok := ret.([]interface{})
+
+	res := []state.QueryItem{}
+
+	arr, ok := ret.([]any)
```
Comment (ItalyPaleAle) [inline state/redis/redis_query.go:301]: Can you extract the pre- and post- in 2 separate functions please, to keep the code more readable

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: tests/certification/state/redis/docker-compose.yml (hunk 1 above)
Comment (JoshVanL) [reply tests/certification/state/redis/docker-compose.yml]: Makes sense, done!

File: state/redis/redis_query.go (hunk 3 above)
Comment (JoshVanL) [reply state/redis/redis_query.go]: Good idea, done :slightly_smiling_face:

**Comment by ItalyPaleAle** [comment]
@JoshVanL code LGTM but looks like the certification test for redis is failing: https://github.com/dapr/components-contrib/actions/runs/7409569254/job/20160102316?pr=3303 can you please take a look?

**Comment by JoshVanL** [comment]
Thanks @ItalyPaleAle, now fixed. Was an issue of `require.Error` vs `require.NoError`

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**